pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -724,181 +722,17 @@ func (k *kubeAPIServer) handleVPNSettings(
 	secretCAVPN *corev1.Secret,
 	secretHTTPProxy *corev1.Secret,
 	secretLegacyVPNCAClient *corev1.Secret,
-	secretLegacyVPNSeed *corev1.Secret,
-	secretLegacyVPNSeedTLSAuth *corev1.Secret,
 	secretHAVPNSeedClient *corev1.Secret,
 	secretHAVPNSeedClientSeedTLSAuth *corev1.Secret,
 ) {
-	if !k.values.VPN.ReversedVPNEnabled {
-		k.handleVPNSettingsLegacy(deployment, secretLegacyVPNCAClient, secretLegacyVPNSeed, secretLegacyVPNSeedTLSAuth)
-	} else if k.values.VPN.HighAvailabilityEnabled {
+
+	if k.values.VPN.HighAvailabilityEnabled {
 		k.handleVPNSettingsHAReversedVPN(deployment, secretCAVPN, secretHAVPNSeedClient, secretHAVPNSeedClientSeedTLSAuth)
 	} else {
 		k.handleVPNSettingsReversedVPN(deployment, configMapEgressSelector, secretCAVPN, secretHTTPProxy)
 	}
 }
 
-func (k *kubeAPIServer) handleVPNSettingsLegacy(
-	deployment *appsv1.Deployment,
-	secretLegacyVPNCAClient *corev1.Secret,
-	secretLegacyVPNSeed *corev1.Secret,
-	secretLegacyVPNSeedTLSAuth *corev1.Secret,
-) {
-	deployment.Spec.Template.Labels[v1beta1constants.LabelNetworkPolicyToShootNetworks] = v1beta1constants.LabelNetworkPolicyAllowed
-	deployment.Spec.Template.Spec.InitContainers = []corev1.Container{{
-		Name:  "set-iptable-rules",
-		Image: k.values.Images.AlpineIPTables,
-		Command: []string{
-			"/bin/sh",
-			"-c",
-			"iptables -A INPUT -i tun0 -p icmp -j ACCEPT && iptables -A INPUT -i tun0 -m state --state NEW -j DROP",
-		},
-		SecurityContext: &corev1.SecurityContext{
-			Capabilities: &corev1.Capabilities{
-				Add: []corev1.Capability{"NET_ADMIN"},
-			},
-			Privileged: pointer.Bool(true),
-		},
-		VolumeMounts: []corev1.VolumeMount{{
-			Name:      volumeNameLibModules,
-			MountPath: volumeMountPathLibModules,
-		}},
-	}}
-	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
-		Name: volumeNameLibModules,
```

Comment:
This const seems no longer used - ref https://github.com/gardener/gardener/blob/44246e18b204925bc8fe782ef978ab1cdfa4066c/pkg/operation/botanist/component/kubeapiserver/deployment.go#L75